api/test/metrics/meter_provider_test.cc

```diff
@@ -39,3 +40,9 @@ TEST(Provider, MultipleMeterProviders)
 
   ASSERT_NE(Provider::GetMeterProvider(), tf);
 }
+#else
+TEST(Provider, DummyTest)
+{
+  // empty
```

Comment:
Why do we need the dummy test if metrics is not enabled in build?

Reply:
It was required as test targets were failing without that. Now I have changed the cmake conf to ignore metrics-specific directories if `WITH_METRICS_PREVIEW` is not set. So could remove these empty tests also